Mergin/validation.py

```diff
@@ -191,9 +198,16 @@ def check_offline(self):
                 dp_name = layer.dataProvider().name()
             except AttributeError:
                 # might be vector tiles - no provider name
-                continue
+                if layer.type() == QgsMapLayerType.VectorTileLayer:
+                    if "type=mbtiles" not in layer.source() and "type=vtpk" not in layer.source():
+                        w.items.append(layer.name())
```

Comment:
I think on QGIS < 3.22 things will not work because `dp_name` will be undefined on the next line.

I would suggest to restructure the whole section for better clarity e.g. like this (not tested):
```python
# special check for vector tile layers because in QGIS < 3.22 they may not have data provider assigned
if layer.type() == QgsMapLayerType.VectorTileLayer:
    # mbtiles/vtpk are always local files
    if "type=mbtiles" in layer.source() or "type=vtpk" in layer.source():
        continue
    w.items.append(layer.name())
    continue

dp_name = layer.dataProvider().name()
if dp_name in QGIS_NET_PROVIDERS + QGIS_DB_PROVIDERS:
    # raster tiles in mbtiles are always local files
    if dp_name == "xyz" and "type=mbtiles" in layer.source():
        continue
    w.items.append(layer.name())
```